doc/Programming-guide/Key-platform-features/Data-collection/index.md

```diff
@@ -5,4 +5,1175 @@ permalink: /:path/
+
+# Basic architecture
+
+The following diagram illustrates basic entities and data flows in scope of the data collection management:
+
+* The data from the endpoints (logs) is collected and transferred to the server in the format as defined by the [log schema](#log-schema) 
+created by the developer for the application
+* Log appenders submit the logs received by the server to a particular storage or analytics system
+
+![Basic data collection management](attach/basic-data-collection-management.png) <br/>
+
+The Kaa Data collection subsystem provides the following features:
+
+* Generation of the logging model and related API calls in the endpoint SDK
+* Enforcement of data integrity and validity
+* Efficient delivery of logs to Operations services
+* Storing log contents by means of the log appenders configured for the application
+
+The application developer is responsible for designing the log schema and invoking the endpoint logging API from the client application.
+
+# Log schema
+
+The log schema is fully compatible with the [Apache Avro schema](http://avro.apache.org/docs/current/spec.html#schemas). There is one log schema defined 
+by default for each Kaa application. This schema supports versioning, therefore, whenever a new log schema is configured on the Kaa service for the application,
+this new schema gets a new sequence version assigned. The Kaa service maintains compatibility with the older versions of the log schema to ensure proper
+functioning of the clients that for some reason are not yet upgraded to the latest schema version.
+
+The following examples illustrate basic log schemas.
+
+* The simplest definition of a log record with no data fields (mostly useless):
+
+```json
+{  
+    "name":"EmptyLog",  
+    "namespace":"org.kaaproject.sample",  
+    "type":"record",  
+    "fields":[
+    ]
+}
+```
+
+* A simple log schema with the log level, tag, and message:
+
+```json
+{
+    "name":"LogData",
+    "namespace":"org.kaaproject.sample",
+    "type":"record",
+    "fields":[
+        {
+            "name":"level",
+            "type":{
+                "type":"enum",
+                "name":"Level",
+                "symbols":[
+                    "DEBUG",
+                    "ERROR",
+                    "FATAL",
+                    "INFO",
+                    "TRACE",
+                    "WARN"
+                ]
+            }
+        },
+        {
+            "name":"tag",
+            "type":"string"
+        },
+        {
+            "name":"message",
+            "type":"string"
+        }
+    ]
+}
+```
+
+## Adding log schema
+
+The default log schema installed for Kaa applications is empty. You can configure your own log schema using the Admin UI or
+[Admin REST API]({{root_url}}Programming-guide/Server-REST-APIs/ #TODO).
+For the purpose of this guide, we will use schema that is very close to the common log structure:
+* log level
```

Comment:
This seems to be a bullet-list by compiled it looks like 'to the common log structure: * log level * tag * message'.
Please fix the formatting if the latter is wrong.